umbrella-feeds-reader/src/lib.rs

```diff
@@ -0,0 +1,266 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+//! # Umbrella feeds
+//!
+//! This is an optional price reader for just one feed.
+//! For maximum gas optimization it is recommended to use UmbrellaFeeds directly.
+//!
+//! This contract has a hard-coded `umbrella_feed` contract address.
+//! ATTENTION: `Umbrella_feed` contract should only be upgraded via the `atomicUpdate` entry point in the registry contract to not break this link.
+use concordium_std::*;
+use core::fmt::Debug;
+
+#[derive(Serialize, SchemaType, Copy, Clone, Debug, PartialOrd, Ord, PartialEq, Eq)]
+pub struct PriceData {
+    /// This is a placeholder, that can be used for some additional data.
+    /// It is only used as marker for removed data (when data == u8::MAX) at the moment.
+    pub data: u8,
+    /// The heartbeat specifies how often the price data will be refreshed in case the price stays flat.
+    /// ATTENTION: u64 is used here instead of u24 (different from the original solidity smart contracts).
+    pub heartbeat: u64,
+    /// It is the time the validators run consensus to decide on the price data.
+    pub timestamp: Timestamp,
+    /// The price.
+    pub price: u128,
+}
+
+#[derive(Serial, Deserial, Debug, SchemaType)]
+struct State {
+    /// Registry contract where the list of all addresses of this protocol is stored.
+    registry: ContractAddress,
+    /// Umbrella_feeds contract where price data is stored.
+    umbrella_feeds: ContractAddress,
+    /// The key for the feed name represented by this contract. E.g. for the "ETH-USD" feed, the key will be the hash("ETH-USD").
+    key: HashSha2256,
```

Comment:
Yes, it is kept for consistency because the Ethereum contracts use a hash of the "ETH-USD" string here. 

Solidity has no good support to deal with strings. Strings are NO native types that exist in solidity. You need to import extra libraries for strings in solidity to do typical string operations, not recommended due to the gas inefficiencies on Ethereum. I assume the umbrella team would like to keep this for easier integration with their backends for MVP. 

@DZariusz

Reply:
yes, gas was main reason for using hash. if this is not the case in Concordium, we can use strings. It will be some additional work on validators side, but if we can save gas, then it is worth it. 
We need to remember that this will change also the way we signing data. We can discuss this on slack or on call.